tasties_app/views.py

```diff
@@ -1,63 +1,84 @@
-from django.shortcuts import render, redirect
-from tasties_app.models import Recipe, Rating
-from django.db.models import Avg
-from collections import OrderedDict
-from django.contrib.auth import logout, login, authenticate
+from django.contrib import messages
+from django.contrib.auth import authenticate, login, logout
 from django.contrib.auth.decorators import login_required
+from django.db.models import Avg
+from django.shortcuts import redirect, render
+from tasties_app.models import Category, Recipe
+
 from .forms import CreateUserForm
-from django.contrib import messages
 
 
 def base(request):
-    return render(request, 'tasties_app/base.html',)
+    return render(
+        request,
+        "tasties_app/base.html",
+    )
 
 
-@login_required(login_url='login')
+@login_required(login_url="login")
 def recipes(request):
-    recipes_list = Recipe.objects.all()
-    recipes_with_ratings = {}
-    for recipe in recipes_list:
-        recipes_with_ratings[recipe] = Rating.objects.filter(recipe_id=recipe).aggregate(Avg('rating'))['rating__avg']
-    recipes_with_ratings = OrderedDict(sorted(recipes_with_ratings.items(), key=lambda x: x[1], reverse=True))
-    context = {'recipes_with_ratings': recipes_with_ratings}
-    return render(request, 'tasties_app/recipes.html', context)
+    recipes_list = Recipe.objects.annotate(
+        recipe_rating=Avg("rating__rating")
+    ).order_by("-recipe_rating")
+    categories_list = Category.objects.all()
+    selected_category = request.GET.get("category")
+    if selected_category:
+        if selected_category != "remove_filter":
+            category = selected_category
+            try:
+                category = Category.objects.get(category_name=selected_category)
+            except Exception:
+                pass
```

Comment:
This will allow users to enter malicious URL addresses and we will still digest it and return a valid page.
I would rather that we will return a forbidden response here because the value of the category must be valid as it's hard-wired to the HTML button element.
And we should test this corner case too.

Reply:
@liornoy @ohadrevach Thanks for the information, didn't think this crossed anyone's mind.